diff --git a/elysia_cron.info b/elysia_cron.info
index 5b05773..c917507 100644
--- a/elysia_cron.info
+++ b/elysia_cron.info
@@ -3,8 +3,7 @@ name = "Elysia Cron"
 description = "Extended cron support with crontab-like scheduling and other features."
 core = 6.x
 ; Information added by drupal.org packaging script on 2009-10-01
-version = "6.x-1.2"
+version = "6.x-1.2-p1"
 core = "6.x"
 project = "elysia_cron"
 datestamp = "1254389732"
-
diff --git a/elysia_cron.module b/elysia_cron.module
index 78de2d4..bcaefce 100644
--- a/elysia_cron.module
+++ b/elysia_cron.module
@@ -506,7 +506,7 @@ function elysia_cron_admin_page() {
         'colspan' => 4)
       );
       $rows[] = array(
-        $conf['rule'], elysia_cron_date($conf['last_run']), $conf['last_execution_time'].'s',
+        check_plain($conf['rule']), elysia_cron_date($conf['last_run']), $conf['last_execution_time'].'s',
         $conf['execution_count'], $conf['avg_execution_time'].'s / '.$conf['max_execution_time'].'s'
        
       );
@@ -898,7 +898,25 @@ function elysia_cron_settings_form_validate($form, $form_state) {
   foreach ($form_values as $key => $value) {
     if ($value && preg_match('/^_elysia_cron_([^_]+_[^_]+)_(.*)$/', $key, $r) && ($r[1] == 'job_rule' || $r[1] == 'ctx_rule')) {
       if (!preg_match('/^\\s*([0-9*,\/-]+[ ]+[0-9*,\/-]+[ ]+[0-9*,\/-]+[ ]+[0-9*,\/-]+[ ]+[0-9*,\/-]+)\\s*$/', $value))
-        form_set_error($key, t('Invalid rule: !rule', array('!rule' => $value)));
+        form_set_error($key, t('Invalid rule: %rule', array('%rule' => $value)));
+    }
+  }
+
+  if (!empty($form_state['values']['elysia_cron_default_rules'])) {
+    $rules = explode(PHP_EOL, $form_state['values']['elysia_cron_default_rules']);
+    foreach ($rules as $rule) {
+      $rule = trim($rule);
+      if (empty($rule)) {
+        continue;
+      }
+
+      $rule = explode('=', $rule);
+      if (empty($rule[1])) {
+        form_set_error('elysia_cron_default_rules', t('Invalid rule: %rule', array('%rule' => $rule[0])));
+      }
+      elseif (!preg_match('/^\\s*([0-9*,\/-]+[ ]+[0-9*,\/-]+[ ]+[0-9*,\/-]+[ ]+[0-9*,\/-]+[ ]+[0-9*,\/-]+)\\s*$/', trim($rule[1]))) {
+        form_set_error('elysia_cron_default_rules', t('Invalid rule: %rule', array('%rule' => $rule[0])));
+      }
     }
   }
 }
