diff --git a/phpass.info b/phpass.info
index 602ff11..0785cc9 100644
--- a/phpass.info
+++ b/phpass.info
@@ -4,8 +4,7 @@ core = 6.x
 php = 5.2.4
 
 ; Information added by Drupal.org packaging script on 2014-11-19
-version = "6.x-2.1"
+version = "6.x-2.1-p1"
 core = "6.x"
 project = "phpass"
 datestamp = "1416425332"
-
diff --git a/phpass.module b/phpass.module
index 335d215..f071610 100644
--- a/phpass.module
+++ b/phpass.module
@@ -15,6 +15,21 @@ function phpass_user($op, &$edit, &$account, $category = NULL) {
       // Catch password changes and update the password hash.
       if (!empty($edit['pass'])) {
         phpass_user_rehash_password($account, $edit['pass']);
+
+        // If the password changed, delete all open sessions and recreate
+        // the current one. The following code is copied from user.module
+        if (is_object($account) && $account->uid) {
+          sess_destroy_uid($account->uid);
+          if ($account->uid == $GLOBALS['user']->uid) {
+            if (function_exists('drupal_session_regenerate')) {
+              // Support for Pressflow.
+              drupal_session_regenerate();
+            }
+            else {
+              sess_regenerate();
+            }
+          }
+        }
       }
       // Prevent the md5 from being saved on update.
       $edit['pass'] = NULL;
