diff --git a/bulk_export/bulk_export.info b/bulk_export/bulk_export.info
index f5afa3f..17464c1 100644
--- a/bulk_export/bulk_export.info
+++ b/bulk_export/bulk_export.info
@@ -4,7 +4,7 @@ core = 6.x
 dependencies[] = ctools
 package = Chaos tool suite
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/ctools.info b/ctools.info
index 1e43d91..42f6a30 100644
--- a/ctools.info
+++ b/ctools.info
@@ -3,7 +3,7 @@ description = A library of helpful tools by Merlin of Chaos.
 core = 6.x
 package = Chaos tool suite
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/ctools_access_ruleset/ctools_access_ruleset.info b/ctools_access_ruleset/ctools_access_ruleset.info
index 344fb20..9884f35 100644
--- a/ctools_access_ruleset/ctools_access_ruleset.info
+++ b/ctools_access_ruleset/ctools_access_ruleset.info
@@ -5,7 +5,7 @@ package = Chaos tool suite
 dependencies[] = ctools
 
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/ctools_ajax_sample/ctools_ajax_sample.info b/ctools_ajax_sample/ctools_ajax_sample.info
index 839c7a1..b9613a9 100644
--- a/ctools_ajax_sample/ctools_ajax_sample.info
+++ b/ctools_ajax_sample/ctools_ajax_sample.info
@@ -5,7 +5,7 @@ dependencies[] = ctools
 version = "6.x-1.0"
 core = "6.x"
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/ctools_custom_content/ctools_custom_content.info b/ctools_custom_content/ctools_custom_content.info
index e7c62ef..4cbce0b 100644
--- a/ctools_custom_content/ctools_custom_content.info
+++ b/ctools_custom_content/ctools_custom_content.info
@@ -5,7 +5,7 @@ package = Chaos tool suite
 dependencies[] = ctools
 
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/ctools_plugin_example/ctools_plugin_example.info b/ctools_plugin_example/ctools_plugin_example.info
index 7aa27db..fca2ad8 100644
--- a/ctools_plugin_example/ctools_plugin_example.info
+++ b/ctools_plugin_example/ctools_plugin_example.info
@@ -8,7 +8,7 @@ dependencies[] = advanced_help
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/js/ajax-responder.js b/js/ajax-responder.js
index 530acc3..44669ed 100644
--- a/js/ajax-responder.js
+++ b/js/ajax-responder.js
@@ -50,7 +50,7 @@
     var $objects = $('a[href="' + old_url + '"]');
     $objects.addClass('ctools-fetching');
     try {
-      var url = Drupal.CTools.AJAX.urlReplaceNojs(url);
+      var url = Drupal.sanitizeAjaxUrl(Drupal.CTools.AJAX.urlReplaceNojs(url));
       var ajaxOptions = {
         type: "POST",
         url: url,
@@ -68,7 +68,8 @@
         complete: function() {
           $objects.removeClass('ctools-fetching');
         },
-        dataType: 'json'
+        dataType: 'json',
+        jsonp: false
       };
       $.ajax(ajaxOptions);
     }
@@ -117,7 +118,7 @@
     }
     $(this).addClass('ctools-ajaxing');
     try {
-      url = Drupal.CTools.AJAX.urlReplaceNojs(url);
+      url = Drupal.sanitizeAjaxUrl(Drupal.CTools.AJAX.urlReplaceNojs(url));
       $.ajax({
         type: "POST",
         url: url,
@@ -131,7 +132,8 @@
         complete: function() {
           $('.ctools-ajaxing').removeClass('ctools-ajaxing');
         },
-        dataType: 'json'
+        dataType: 'json',
+        jsonp: false
       });
     }
     catch (err) {
@@ -159,7 +161,7 @@
     $(this).addClass('ctools-ajaxing');
     try {
       if (url) {
-        url = Drupal.CTools.AJAX.urlReplaceNojs(url);
+        url = Drupal.sanitizeAjaxUrl(Drupal.CTools.AJAX.urlReplaceNojs(url));
         $.ajax({
           type: "POST",
           url: url,
@@ -173,7 +175,8 @@
           complete: function() {
             $('.ctools-ajaxing').removeClass('ctools-ajaxing');
           },
-          dataType: 'json'
+          dataType: 'json',
+          jsonp: false
         });
       }
       else {
@@ -329,7 +332,7 @@
     var form_id = $object.parents('form').get(0).id;
     try {
       if (url) {
-        url = Drupal.CTools.AJAX.urlReplaceNojs(url);
+        url = Drupal.sanitizeAjaxUrl(Drupal.CTools.AJAX.urlReplaceNojs(url));
         $.ajax({
           type: "POST",
           url: url,
@@ -346,7 +349,8 @@
               $('form#' + form_id).submit();
             }
           },
-          dataType: 'json'
+          dataType: 'json',
+          jsonp: false
         });
       }
       else {
diff --git a/page_manager/page_manager.info b/page_manager/page_manager.info
index c7829d9..c6ef776 100644
--- a/page_manager/page_manager.info
+++ b/page_manager/page_manager.info
@@ -5,7 +5,7 @@ dependencies[] = ctools
 package = Chaos tool suite
 
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/stylizer/stylizer.info b/stylizer/stylizer.info
index 055aa56..e22d70a 100644
--- a/stylizer/stylizer.info
+++ b/stylizer/stylizer.info
@@ -5,7 +5,7 @@ package = Chaos tool suite
 dependencies[] = ctools
 
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
diff --git a/views_content/views_content.info b/views_content/views_content.info
index 7bd85ea..ccbf5c5 100644
--- a/views_content/views_content.info
+++ b/views_content/views_content.info
@@ -6,7 +6,7 @@ dependencies[] = views
 core = 6.x
 package = Chaos tool suite
 ; Information added by Drupal.org packaging script on 2015-12-22
-version = "6.x-1.15"
+version = "6.x-1.15-p1"
 core = "6.x"
 project = "ctools"
 datestamp = "1450810742"
