diff -ru xmlsitemap/xmlsitemap.info xmlsitemap/xmlsitemap.info
--- xmlsitemap/xmlsitemap.info	2014-12-05 03:23:25.000000000 +0100
+++ xmlsitemap/xmlsitemap.info	2016-05-04 01:28:25.453283153 +0200
@@ -19,7 +19,7 @@
 configure = admin/config/search/xmlsitemap
 
 ; Information added by Drupal.org packaging script on 2014-12-05
-version = "6.x-2.1"
+version = "6.x-2.1-p1"
 core = "6.x"
 project = "xmlsitemap"
 datestamp = "1417746205"
diff -ru xmlsitemap/xsl/xmlsitemap.xsl.js xmlsitemap/xsl/xmlsitemap.xsl.js
--- xmlsitemap/xsl/xmlsitemap.xsl.js	2014-12-05 03:20:24.000000000 +0100
+++ xmlsitemap/xsl/xmlsitemap.xsl.js	2016-05-04 01:26:15.718408342 +0200
@@ -28,9 +28,10 @@
 });
 
 $(document).ready(function() {
-  // Set some location variales.
-  $('h1').append(': ' + location);
-  document.title += ': ' + location;
+  // Set some location variables.
+  var h1 = $('h1');
+  h1.text(h1.text() + ': ' + location);
+  document.title = h1.text();
 
   $('table').tablesorter({
     sortList: [[0,0]],
