diff --git a/googleanalytics.admin.inc b/googleanalytics.admin.inc
index 1c33278..d4d053d 100644
--- a/googleanalytics.admin.inc
+++ b/googleanalytics.admin.inc
@@ -267,6 +267,8 @@ function googleanalytics_admin_settings_form(&$form_state) {
   );
 
   // Advanced feature configurations.
+  $user_access_add_js_snippets = !user_access('add JS snippets for google analytics');
+  $user_access_add_js_snippets_permission_warning = $user_access_add_js_snippets ? ' <em>' . t('This field has been disabled because you do not have sufficient permissions to edit it.') . '</em>' : '';
   $form['advanced'] = array(
     '#type' => 'fieldset',
     '#title' => t('Advanced settings'),
@@ -332,17 +334,19 @@ function googleanalytics_admin_settings_form(&$form_state) {
     '#type' => 'textarea',
     '#title' => t('Code snippet (before)'),
     '#default_value' => variable_get('googleanalytics_codesnippet_before', ''),
+    '#disabled' => $user_access_add_js_snippets,
     '#rows' => 5,
     '#wysiwyg' => FALSE,
-    '#description' => t("Code in this textarea will be added <strong>before</strong> _gaq.push(['_trackPageview'])."),
+    '#description' => t("Code in this textarea will be added <strong>before</strong> _gaq.push(['_trackPageview']).") . $user_access_add_js_snippets_permission_warning,
   );
   $form['advanced']['codesnippet']['googleanalytics_codesnippet_after'] = array(
     '#type' => 'textarea',
     '#title' => t('Code snippet (after)'),
     '#default_value' => variable_get('googleanalytics_codesnippet_after', ''),
+    '#disabled' => $user_access_add_js_snippets,
     '#rows' => 5,
     '#wysiwyg' => FALSE,
-    '#description' => t("Code in this textarea will be added <strong>after</strong> _gaq.push(['_trackPageview']). This is useful if you'd like to track a site in two accounts."),
+    '#description' => t("Code in this textarea will be added <strong>after</strong> _gaq.push(['_trackPageview']). This is useful if you'd like to track a site in two accounts.") . $user_access_add_js_snippets_permission_warning,
   );
 
   /* hook_footer() is not able to add code to head. Upgrade to D7 required.
diff --git a/googleanalytics.info b/googleanalytics.info
index 06b0f8a..1acaa3f 100644
--- a/googleanalytics.info
+++ b/googleanalytics.info
@@ -4,8 +4,7 @@ core = 6.x
 package = Statistics
 
 ; Information added by drupal.org packaging script on 2013-10-17
-version = "6.x-3.6"
+version = "6.x-3.6-p1"
 core = "6.x"
 project = "google_analytics"
 datestamp = "1382021061"
-
diff --git a/googleanalytics.module b/googleanalytics.module
index 4da6893..5959ba1 100644
--- a/googleanalytics.module
+++ b/googleanalytics.module
@@ -39,7 +39,7 @@ function googleanalytics_theme() {
  * Implementation of hook_perm().
  */
 function googleanalytics_perm() {
-  return array('administer google analytics', 'opt-in or out of tracking', 'use PHP for tracking visibility');
+  return array('administer google analytics', 'opt-in or out of tracking', 'use PHP for tracking visibility', 'add JS snippets for google analytics');
 }
 
 /**
