diff --git a/CHANGELOG.txt b/CHANGELOG.txt
index e6b38ab..86c28b1 100644
--- a/CHANGELOG.txt
+++ b/CHANGELOG.txt
@@ -1,3 +1,18 @@
+Drupal 6.38-p6, 2018-04-17 - SA-CORE-2018-006
+
+The path module allows users with the 'administer url aliases' permission to
+create pretty URLs for content.
+
+In certain circumstances the user can enter a particular path that triggers an
+open redirect to a malicious url.
+
+The issue is mitigated by the fact that the user needs the administer url
+aliases permission to exploit.
+
+Upstream reference:
+https://www.drupal.org/sa-core-2018-006
+
+
 Drupal 6.38-p5, 2018-04-25 - SA-CORE-2018-004
 
 A remote code execution vulnerability may exist within multiple subsystems of
diff --git a/includes/common.inc b/includes/common.inc
index a5c362d..9431bc8 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -1558,6 +1558,10 @@ function url($path = NULL, $options = array()) {
   }
   elseif (!empty($path) && !$options['alias']) {
     $path = drupal_get_path_alias($path, isset($options['language']) ? $options['language']->language : '');
+    // Strip leading slashes from internal paths to prevent them becoming external
+    // URLs without protocol. /example.com should not be turned into
+    // //example.com.
+    $path = ltrim($path, '/');
   }
 
   if (function_exists('custom_url_rewrite_outbound')) {
diff --git a/modules/aggregator/aggregator.info b/modules/aggregator/aggregator.info
index 987dc09..2daff4d 100644
--- a/modules/aggregator/aggregator.info
+++ b/modules/aggregator/aggregator.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/block/block.info b/modules/block/block.info
index 66c95f2..8f8fd2f 100644
--- a/modules/block/block.info
+++ b/modules/block/block.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/blog/blog.info b/modules/blog/blog.info
index 8fa2cc7..2cb7fe7 100644
--- a/modules/blog/blog.info
+++ b/modules/blog/blog.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/blogapi/blogapi.info b/modules/blogapi/blogapi.info
index 6d10dfe..80fe09b 100644
--- a/modules/blogapi/blogapi.info
+++ b/modules/blogapi/blogapi.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/book/book.info b/modules/book/book.info
index ee2c431..addfcc3 100644
--- a/modules/book/book.info
+++ b/modules/book/book.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/color/color.info b/modules/color/color.info
index d8b8dbb..b9ab0cb 100644
--- a/modules/color/color.info
+++ b/modules/color/color.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/comment/comment.info b/modules/comment/comment.info
index ef65654..9122ba8 100644
--- a/modules/comment/comment.info
+++ b/modules/comment/comment.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/contact/contact.info b/modules/contact/contact.info
index 0734d08..baee46f 100644
--- a/modules/contact/contact.info
+++ b/modules/contact/contact.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/dblog/dblog.info b/modules/dblog/dblog.info
index 3a1bbe5..245aaa4 100644
--- a/modules/dblog/dblog.info
+++ b/modules/dblog/dblog.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/filter/filter.info b/modules/filter/filter.info
index c980f69..5ae90c9 100644
--- a/modules/filter/filter.info
+++ b/modules/filter/filter.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/forum/forum.info b/modules/forum/forum.info
index 1a0cfd2..fe24994 100644
--- a/modules/forum/forum.info
+++ b/modules/forum/forum.info
@@ -7,7 +7,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/help/help.info b/modules/help/help.info
index 517eeac..404417e 100644
--- a/modules/help/help.info
+++ b/modules/help/help.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/locale/locale.info b/modules/locale/locale.info
index 300c7ab..673a59c 100644
--- a/modules/locale/locale.info
+++ b/modules/locale/locale.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/menu/menu.info b/modules/menu/menu.info
index ab80655..abccf5b 100644
--- a/modules/menu/menu.info
+++ b/modules/menu/menu.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/node/node.info b/modules/node/node.info
index 973d35b..a12804f 100644
--- a/modules/node/node.info
+++ b/modules/node/node.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/openid/openid.info b/modules/openid/openid.info
index a63b0d8..f2661cf 100644
--- a/modules/openid/openid.info
+++ b/modules/openid/openid.info
@@ -5,7 +5,7 @@ package = Core - optional
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/path/path.info b/modules/path/path.info
index 507fa1c..262653c 100644
--- a/modules/path/path.info
+++ b/modules/path/path.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/php/php.info b/modules/php/php.info
index 933a714..b0fce64 100644
--- a/modules/php/php.info
+++ b/modules/php/php.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/ping/ping.info b/modules/ping/ping.info
index 8a2dde3..9757542 100644
--- a/modules/ping/ping.info
+++ b/modules/ping/ping.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/poll/poll.info b/modules/poll/poll.info
index fe3890e..834aa11 100644
--- a/modules/poll/poll.info
+++ b/modules/poll/poll.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/profile/profile.info b/modules/profile/profile.info
index 6d6c41a..7720a56 100644
--- a/modules/profile/profile.info
+++ b/modules/profile/profile.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/search/search.info b/modules/search/search.info
index 33ae918..b87f19c 100644
--- a/modules/search/search.info
+++ b/modules/search/search.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/statistics/statistics.info b/modules/statistics/statistics.info
index 5649430..72c0659 100644
--- a/modules/statistics/statistics.info
+++ b/modules/statistics/statistics.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/syslog/syslog.info b/modules/syslog/syslog.info
index 3666b33..7cc1219 100644
--- a/modules/syslog/syslog.info
+++ b/modules/syslog/syslog.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/system/system.info b/modules/system/system.info
index 2aac76a..4dd9fcb 100644
--- a/modules/system/system.info
+++ b/modules/system/system.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/taxonomy/taxonomy.info b/modules/taxonomy/taxonomy.info
index e5a0390..0ef1bec 100644
--- a/modules/taxonomy/taxonomy.info
+++ b/modules/taxonomy/taxonomy.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/throttle/throttle.info b/modules/throttle/throttle.info
index 4780dac..e51d45d 100644
--- a/modules/throttle/throttle.info
+++ b/modules/throttle/throttle.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/tracker/tracker.info b/modules/tracker/tracker.info
index e9ecad3..3c51716 100644
--- a/modules/tracker/tracker.info
+++ b/modules/tracker/tracker.info
@@ -6,7 +6,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/translation/translation.info b/modules/translation/translation.info
index 7e6ef73..5a1417a 100644
--- a/modules/translation/translation.info
+++ b/modules/translation/translation.info
@@ -6,7 +6,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/trigger/trigger.info b/modules/trigger/trigger.info
index 45f8705..a861c54 100644
--- a/modules/trigger/trigger.info
+++ b/modules/trigger/trigger.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/update/update.info b/modules/update/update.info
index 73a2252..727c3b0 100644
--- a/modules/update/update.info
+++ b/modules/update/update.info
@@ -5,7 +5,7 @@ package = Core - optional
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/upload/upload.info b/modules/upload/upload.info
index 4be528b..27e119a 100644
--- a/modules/upload/upload.info
+++ b/modules/upload/upload.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/modules/user/user.info b/modules/user/user.info
index dd78112..2694626 100644
--- a/modules/user/user.info
+++ b/modules/user/user.info
@@ -5,7 +5,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/themes/bluemarine/bluemarine.info b/themes/bluemarine/bluemarine.info
index fdf215c..3afb7a8 100644
--- a/themes/bluemarine/bluemarine.info
+++ b/themes/bluemarine/bluemarine.info
@@ -5,7 +5,7 @@ core = 6.x
 engine = phptemplate
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/themes/chameleon/chameleon.info b/themes/chameleon/chameleon.info
index 229d905..9eeaf3a 100644
--- a/themes/chameleon/chameleon.info
+++ b/themes/chameleon/chameleon.info
@@ -12,7 +12,7 @@ version = VERSION
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/themes/chameleon/marvin/marvin.info b/themes/chameleon/marvin/marvin.info
index 663a2af..16c1238 100644
--- a/themes/chameleon/marvin/marvin.info
+++ b/themes/chameleon/marvin/marvin.info
@@ -7,7 +7,7 @@ core = 6.x
 base theme = chameleon
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/themes/garland/garland.info b/themes/garland/garland.info
index 788a29b..44d7054 100644
--- a/themes/garland/garland.info
+++ b/themes/garland/garland.info
@@ -7,7 +7,7 @@ stylesheets[all][] = style.css
 stylesheets[print][] = print.css
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/themes/garland/minnelli/minnelli.info b/themes/garland/minnelli/minnelli.info
index 039c638..37c9522 100644
--- a/themes/garland/minnelli/minnelli.info
+++ b/themes/garland/minnelli/minnelli.info
@@ -6,7 +6,7 @@ base theme = garland
 stylesheets[all][] = minnelli.css
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
diff --git a/themes/pushbutton/pushbutton.info b/themes/pushbutton/pushbutton.info
index ba4ba54..4663a84 100644
--- a/themes/pushbutton/pushbutton.info
+++ b/themes/pushbutton/pushbutton.info
@@ -5,7 +5,7 @@ core = 6.x
 engine = phptemplate
 
 ; Information added by Drupal.org packaging script on 2016-02-24
-version = "6.38-p5"
+version = "6.38-p6"
 project = "drupal"
 datestamp = "1456343372"
 
