diff -ru ./includes/view.inc ./includes/view.inc
--- ./includes/view.inc	2015-02-11 10:41:33.000000000 +0100
+++ ./includes/view.inc	2016-04-20 18:17:51.550523338 +0200
@@ -416,7 +416,7 @@
    */
   function _init_handler($key, $info) {
     // Load the requested items from the display onto the object.
-    $this->$key = $this->display_handler->get_handlers($key);
+    $this->$key = &$this->display_handler->get_handlers($key);
 
     // This reference deals with difficult PHP indirection.
     $handlers = &$this->$key;
Nur in ./includes: view.inc.rej.
diff -ru /tmp/views-6.x-2.18/plugins/views_plugin_display.inc ./plugins/views_plugin_display.inc
--- ./plugins/views_plugin_display.inc	2015-02-11 10:41:33.000000000 +0100
+++ ./plugins/views_plugin_display.inc	2016-04-20 18:17:51.550523338 +0200
@@ -565,7 +565,7 @@
   /**
    * Get a full array of handlers for $type. This caches them.
    */
-  function get_handlers($type) {
+  function &get_handlers($type) {
     if (!isset($this->handlers[$type])) {
       $this->handlers[$type] = array();
       $types = views_object_types();
Nur in .: tag1-views-6.2-drupal-sa-contrib-2014-054.patch.
diff -ru /tmp/views-6.x-2.18/views.info ./views.info
--- ./views.info	2015-02-11 10:43:13.000000000 +0100
+++ ./views.info	2016-04-20 18:04:52.484779050 +0200
@@ -4,7 +4,7 @@
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2015-02-11
-version = "6.x-2.18"
+version = "6.x-2.18-p1"
 core = "6.x"
 project = "views"
 datestamp = "1423647793"
