diff --git a/views.info b/views.info
index 9f5dc4f..b984881 100644
--- a/views.info
+++ b/views.info
@@ -4,7 +4,7 @@ package = Views
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2015-02-11
-version = "6.x-3.2-p1"
+version = "6.x-3.2-p2"
 core = "6.x"
 project = "views"
 datestamp = "1423647787"
diff --git a/views_export/views_export.info b/views_export/views_export.info
index f855501..4f3fe5f 100644
--- a/views_export/views_export.info
+++ b/views_export/views_export.info
@@ -6,7 +6,7 @@ dependencies[] = views
 core = 6.x
 
 ; Information added by Drupal.org packaging script on 2015-02-11
-version = "6.x-3.2"
+version = "6.x-3.2-p2"
 core = "6.x"
 project = "views"
 datestamp = "1423647787"
diff --git a/views_ui.info b/views_ui.info
index 60ea7ee..f4b0282 100644
--- a/views_ui.info
+++ b/views_ui.info
@@ -5,7 +5,7 @@ core = 6.x
 dependencies[] = views
 
 ; Information added by Drupal.org packaging script on 2015-02-11
-version = "6.x-3.2"
+version = "6.x-3.2-p2"
 core = "6.x"
 project = "views"
 datestamp = "1423647787"
